cmd/mapstructure-to-hcl2/mapstructure-to-hcl2.go

```diff
@@ -451,19 +451,31 @@ func getMapstructureSquashedStruct(topPkg *types.Package, utStruct *types.Struct
 			log.Printf("could not parse field tag %s of : %v", tag, err)
 			continue
 		}
-		if ms, err := structtag.Get("mapstructure"); err != nil {
-			//no mapstructure tag
-		} else if ms.HasOption("squash") {
-			ot := field.Type()
-			uot := ot.Underlying()
-			utStruct, utOk := uot.(*types.Struct)
-			if !utOk {
+
+		// Contains mapstructure-to-hcl2 tag
+		if ms, err := structtag.Get("mapstructure-to-hcl2"); err == nil {
+			// Stop if is telling to skip it
+			if ms.HasOption("skip") {
```

Comment:
I added only the `skip` option because it's the only one we have right now. If we start using `mapstructure-to-hcl2`with `squash` we can add it later.

Reply:
Super nice, I think it's good just like this !